Py_FS/wrapper/nature_inspired/AOA.py

```diff
@@ -0,0 +1,134 @@
+"""
+Programmer: Soumitri Chattopadhyay
+Date of Development: 11/07/2021
+This code has been developed according to the procedures mentioned in the following research article:
+"Laith A., Diabat A., Mirjalili S., Elaziz M.A., Gandomi A.H. The Arithmetic Optimization Algorithm.
+Computer Methods in Applied Mechanics and Engineering, 376, 113609 (2021)"
+"""
+
+import math
+
+import numpy as np
+from sklearn import datasets
+
+from Py_FS.wrapper.nature_inspired.algorithm import Algorithm
+
+
+class AOA(Algorithm):
+
+    # Arithmetic Optimization Algorithm
+    ############################### Parameters ####################################
+    #                                                                             #
+    #   num_agents: number of agents                                              #
+    #   max_iter: maximum number of generations                                   #
+    #   train_data: training samples of data                                      #
+    #   train_label: class labels for the training samples                        #
+    #   obj_function: the function to maximize while doing feature selection      #
+    #   trans_function_shape: shape of the transfer function used                 #
+    #   save_conv_graph: boolean value for saving convergence graph               #
+    #                                                                             #
+    ###############################################################################
+
+    def __init__(self,
+                 num_agents,
+                 max_iter,
+                 train_data,
+                 train_label,
+                 save_conv_graph=False,
+                 trans_func_shape='s',
+                 seed=0):
+
+        super().__init__(num_agents=num_agents,
+                         max_iter=max_iter,
+                         train_data=train_data,
+                         train_label=train_label,
+                         save_conv_graph=save_conv_graph,
+                         trans_func_shape=trans_func_shape,
+                         seed=seed)
+
+        self.algo_name = 'AOA'
+        self.agent_name = 'Agent'
+        self.algo_params = {}
+
+    def user_input(self):
+        # initializing parameters
+        self.algo_params['Min'] = 0.1
+        self.algo_params['Max'] = 0.9
+        self.algo_params['EPS'] = 1e-6
+        self.algo_params['alpha'] = 5
+        self.algo_params['mu'] = 0.5
+
```

Comment:
Allow the user to enter the values of the parameters [Refer to the user_input() method of WOA_test.py].

Reply:
Done, ranges/default values given as in relevant paper